lib/Dialect/Torch/Utils/Utils.cpp

```diff
@@ -525,3 +525,33 @@ LogicalResult Torch::getTransposedType(BaseTensorType inType, int64_t dimA,
                                                inType.getOptionalDtype());
   return success();
 }
+
+Type Torch::getDefaultAccType(PatternRewriter &rewriter, Type inputType) {
+  if (inputType.isF16())
+    return rewriter.getF32Type();
+  if (inputType.isBF16())
+    return rewriter.getF32Type();
+  if (inputType.isa<Float32Type>())
+    return rewriter.getF32Type();
+  if (inputType.isa<Float64Type>())
+    return rewriter.getF64Type();
+  if (inputType.isFloat8E5M2())
+    return rewriter.getF32Type();
+  if (inputType.isFloat8E4M3FN())
+    return rewriter.getF32Type();
+  if (inputType.isFloat8E5M2FNUZ())
+    return rewriter.getF32Type();
+  if (inputType.isFloat8E4M3FNUZ())
+    return rewriter.getF32Type();
+  if (inputType.isSignedInteger(8))
+    return rewriter.getI64Type();
```

Comment:
This is not correct, `int8` is almost guaranteed to be an `int32_t` accumulator

Reply:
I'm following the convention from CUDA - https://github.com/pytorch/pytorch/blob/862b99b571e7544c82232aa6956a8ddd0ab247ed/aten/src/ATen/AccumulateType.h#L118

Reply:
This is strange but we should match. We will probably want a linalg lowering that truncates the accumulator type when it is guaranteed to fall within bounds.